download/src/lib.rs

```diff
@@ -345,6 +345,10 @@ pub mod reqwest_be {
 
     fn client_generic() -> ClientBuilder {
         Client::builder()
+            // HACK: set `pool_max_idle_per_host` to `0` to avoid an issue in the underlying
```

Comment:
Can you explain why this is necessary/why there is no better solution for this?

Reply:
The underlying issue is https://github.com/hyperium/hyper/issues/2312, see https://github.com/hyperium/hyper/issues/2312#issuecomment-778005053 for the first occurrence of this hack. It is suspected that there is a hyper pool deadlock underneath, and this workaround has been used by many to successfully mitigate the issue.

Reply:
Do we have a good reproduction scenario?

Reply:
@djc Unfortunately not yet, and that's the tricky part of this thing I guess... If it really is a concurrency bug, then it'll be rather hard to track and to fix (that's also why that issue has been open for such a long time I guess, a previous project that I collaborated on has also hit it back in 2020 🤔).

As far as our CI is concerned, we can see several failures in the past 5 days caused by this one, and they are all "solved" by relaunching the test task. The particular test(s) that failed varied from time to time, but I'm sure that all those tests are `reqwest`-related.

Reply:
@djc The workaround seems to be working: our CI hasn't been failing for this reason in the past 5 days.